kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11584413?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11584413?buildTab=artifacts#/) on release-23.1 @ [d762d323b40b09feba4604e5c9c475614fd21b0c](https://github.com/cockroachdb/cockroach/commits/d762d323b40b09feba4604e5c9c475614fd21b0c):

Fatal error:

```
panic: priority range ID already set: old=2, new=66, first set at:

```
Stack: 

```
goroutine 20051967 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc0094e26d0, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x35
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc0057fe628?, 0xc013bfa6a8?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x85
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc0057fe580, {0x5f3e588, 0xc00ff82990}, 0xc00de92ee0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0x867
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc0057fe580, {0x1, 0xa, {0xa, 0x0, 0xc00de92ee0, 0xc017d50140, 0xc007583400, 0xc0075833a0, 0xc012f29cc0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x1fd
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc00de92ee0, 0xc017d50140, 0xc007583400, 0xc0075833a0, 0xc012f29cc0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1c5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc017bbf500, {0x5f3e588, 0xc004dcc330}, 0xc00e783480)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x114a
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc01348ea80, {0x5f3e588, _}, {_, _}, {_, _}, {_, _}, {0x5f181e0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0xfb9
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc001c3a000, {0x5f3e518?, 0xc000084068?})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x1f76
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc004930600, {0x5f3e518, 0xc000084068})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x36
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x5f01e80, 0xc00492e600}, {0x5f014c0, 0xc00ae303c0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x143
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc01049dd40)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x265
testing.tRunner(0xc01049dd40, 0x4f34550)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
 [recovered]
	panic: priority range ID already set: old=2, new=66, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries2007773172
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_put
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_put_after_timestamp_leaked
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31128

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11584414?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11584414?buildTab=artifacts#/) on release-23.1 @ [d762d323b40b09feba4604e5c9c475614fd21b0c](https://github.com/cockroachdb/cockroach/commits/d762d323b40b09feba4604e5c9c475614fd21b0c):

Fatal error:

```
panic: priority range ID already set: old=2, new=64, first set at:

```
Stack: 

```
goroutine 96267 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc004fb2ab8, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc00e21e300, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x9b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc0084f5100, {0xa722708, 0xc00a8f5e60}, 0xc0034b0bd0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc0084f5100, {0x1, 0xa, {0xa, 0x0, 0xc0034b0bd0, 0xc00709c190, 0xc00003bd50, 0xc006a79fd0, 0xc00102cc80, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x2ae
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc0034b0bd0, 0xc00709c190, 0xc00003bd50, 0xc006a79fd0, 0xc00102cc80, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1a5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc006f4ea80, {0xa722708, 0xc00b07d7a0}, 0xc005087170)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x1f65
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc006c37500, {0xa722708, _}, {_, _}, {_, _}, {_, _}, {0xa6fc3c0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0x167a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc003036000, {0xa722698, 0xc0001b4008})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x33db
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc005ab8c00, {0xa722698, 0xc0001b4008})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x90
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0xa6e6040, 0xc007ad8c00}, {0xa6e5680, 0xc0064b2af0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc005450000)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x477
testing.tRunner(0xc005450000, 0x7727908)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=64, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries2439893080
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11599972?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11599972?buildTab=artifacts#/) on release-23.1 @ [58ac87a9b24e1cfb446143dafe04808ea186637f](https://github.com/cockroachdb/cockroach/commits/58ac87a9b24e1cfb446143dafe04808ea186637f):

Fatal error:

```
panic: priority range ID already set: old=2, new=65, first set at:

```
Stack: 

```
goroutine 20156771 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc011194910, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x35
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc005722d28?, 0xc004fb8510?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x85
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc005722c80, {0x5f3e5e8, 0xc00b26bad0}, 0xc008800310)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0x867
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc005722c80, {0x1, 0xa, {0xa, 0x0, 0xc008800310, 0xc0010d37c0, 0xc0045d55f0, 0xc0045d5540, 0xc00ef297c0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x1fd
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc008800310, 0xc0010d37c0, 0xc0045d55f0, 0xc0045d5540, 0xc00ef297c0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1c5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc0050dc000, {0x5f3e5e8, 0xc0085ee270}, 0xc00c682580)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x114a
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc007962a80, {0x5f3e5e8, _}, {_, _}, {_, _}, {_, _}, {0x5f18240, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0xfb9
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc007963500, {0x5f3e578?, 0xc000084068?})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x1f76
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc009a30000, {0x5f3e578, 0xc000084068})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x36
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x5f01ee0, 0xc00f8fc300}, {0x5f01520, 0xc005be1770}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x143
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc00fe34820)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x265
testing.tRunner(0xc00fe34820, 0x4f345b0)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
 [recovered]
	panic: priority range ID already set: old=2, new=65, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries3474843821
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_cput_after_timestamp_leaked
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_put_after_timestamp_leaked
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_cput_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_put
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_before_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_put_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_cput
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_put_and_scan_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_scan_and_cput
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_scan_before_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_scan_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_delete_range
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_initput
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_put_and_get_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_too_many_refreshes,_read_only
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11599973?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11599973?buildTab=artifacts#/) on release-23.1 @ [58ac87a9b24e1cfb446143dafe04808ea186637f](https://github.com/cockroachdb/cockroach/commits/58ac87a9b24e1cfb446143dafe04808ea186637f):

Fatal error:

```
panic: priority range ID already set: old=2, new=64, first set at:

```
Stack: 

```
goroutine 101178 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc000d7faf8, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc008f05380, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x9b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc004e67b00, {0xa7226c8, 0xc001080e70}, 0xc004867b20)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc004e67b00, {0x1, 0xa, {0xa, 0x0, 0xc004867b20, 0xc00552e280, 0xc0021691f0, 0xc0021690e0, 0xc0016bc000, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x2ae
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc004867b20, 0xc00552e280, 0xc0021691f0, 0xc0021690e0, 0xc0016bc000, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1a5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc008223500, {0xa7226c8, 0xc0058a19b0}, 0xc004971560)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x1f65
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc003fc0a80, {0xa7226c8, _}, {_, _}, {_, _}, {_, _}, {0xa6fc380, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0x167a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc003fc1500, {0xa722658, 0xc0000820b8})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x33db
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc004450600, {0xa722658, 0xc0000820b8})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x90
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0xa6e6000, 0xc00a396900}, {0xa6e5640, 0xc009d6ee10}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc008220820)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x477
testing.tRunner(0xc008220820, 0x7727888)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=64, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries3925252608
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11608966?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11608966?buildTab=artifacts#/) on release-23.1 @ [52956dd5bb56962342ea2e9db6dcbd1dbcf864f7](https://github.com/cockroachdb/cockroach/commits/52956dd5bb56962342ea2e9db6dcbd1dbcf864f7):

Fatal error:

```
panic: priority range ID already set: old=2, new=65, first set at:

```
Stack: 

```
goroutine 20028679 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc0097ec2e0, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x35
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc00a6d92a8?, 0xc02257fae8?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x85
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc00a6d9200, {0x5f3e5e8, 0xc01ea64ab0}, 0xc023ac38f0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0x867
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc00a6d9200, {0x1, 0xa, {0xa, 0x0, 0xc023ac38f0, 0xc02441fa40, 0xc00bdd5e70, 0xc00bdd5e10, 0xc002c5e320, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x1fd
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc023ac38f0, 0xc02441fa40, 0xc00bdd5e70, 0xc00bdd5e10, 0xc002c5e320, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1c5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc02480aa80, {0x5f3e5e8, 0xc004cd8f00}, 0xc016244700)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x114a
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc000fff500, {0x5f3e5e8, _}, {_, _}, {_, _}, {_, _}, {0x5f18240, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0xfb9
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc011d7ea80, {0x5f3e578?, 0xc0001ac000?})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x1f76
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc01e55e600, {0x5f3e578, 0xc0001ac000})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x36
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x5f01ee0, 0xc01e8b6600}, {0x5f01520, 0xc004b1d180}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x143
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc01e16d380)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x265
testing.tRunner(0xc01e16d380, 0x4f345b0)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
 [recovered]
	panic: priority range ID already set: old=2, new=65, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries2344210490
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_delete_range
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_scan_and_cput
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_scan_before_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_too_many_refreshes,_read_only
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11608970?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11608970?buildTab=artifacts#/) on release-23.1 @ [52956dd5bb56962342ea2e9db6dcbd1dbcf864f7](https://github.com/cockroachdb/cockroach/commits/52956dd5bb56962342ea2e9db6dcbd1dbcf864f7):

Fatal error:

```
panic: priority range ID already set: old=2, new=64, first set at:

```
Stack: 

```
goroutine 95932 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc0097ac478, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc0008a6780, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x9b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc00537bb00, {0xa7226c8, 0xc00b94e240}, 0xc00085fe30)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc00537bb00, {0x1, 0xa, {0xa, 0x0, 0xc00085fe30, 0xc004d75c20, 0xc0031ac790, 0xc0031ac660, 0xc0025c3ae0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x2ae
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc00085fe30, 0xc004d75c20, 0xc0031ac790, 0xc0031ac660, 0xc0025c3ae0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1a5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc0045caa80, {0xa7226c8, 0xc00c6ef290}, 0xc001bf8900)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x1f65
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc0035d0a80, {0xa7226c8, _}, {_, _}, {_, _}, {_, _}, {0xa6fc380, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0x167a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc001c2aa80, {0xa722658, 0xc0001b4008})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x33db
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc002b88c00, {0xa722658, 0xc0001b4008})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x90
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0xa6e6000, 0xc005d58f00}, {0xa6e5640, 0xc003731090}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc0062d4000)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x477
testing.tRunner(0xc0062d4000, 0x7727888)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=64, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries657377977
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109977 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1.10-rc release-blocker]
- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11618947?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11618947?buildTab=artifacts#/) on release-23.1 @ [6e34fffddbb0a88612b712ec5d41e4781d130bab](https://github.com/cockroachdb/cockroach/commits/6e34fffddbb0a88612b712ec5d41e4781d130bab):

Fatal error:

```
panic: priority range ID already set: old=2, new=65, first set at:

```
Stack: 

```
goroutine 20002976 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc0098ee130, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x35
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc0030412a8?, 0xc01f942e28?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x85
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc003041200, {0x5f3eae8, 0xc01592fb60}, 0xc012b918f0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0x867
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc003041200, {0x1, 0xa, {0xa, 0x0, 0xc012b918f0, 0xc00c2a3770, 0xc0033246c0, 0xc003324640, 0xc00a0b4140, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x1fd
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc012b918f0, 0xc00c2a3770, 0xc0033246c0, 0xc003324640, 0xc00a0b4140, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1c5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc01774c000, {0x5f3eae8, 0xc010c500f0}, 0xc0183b4500)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x114a
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc002354a80, {0x5f3eae8, _}, {_, _}, {_, _}, {_, _}, {0x5f18740, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0xfb9
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc002355500, {0x5f3ea78?, 0xc000084068?})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x1f76
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc010960000, {0x5f3ea78, 0xc000084068})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x36
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x5f023e0, 0xc01238a900}, {0x5f01a20, 0xc0171e0410}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x143
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc015e1eea0)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x265
testing.tRunner(0xc015e1eea0, 0x4f34990)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
 [recovered]
	panic: priority range ID already set: old=2, new=65, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries3370871620
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_delete_range
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109977 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1.10-rc release-blocker]
- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11618954?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11618954?buildTab=artifacts#/) on release-23.1 @ [6e34fffddbb0a88612b712ec5d41e4781d130bab](https://github.com/cockroachdb/cockroach/commits/6e34fffddbb0a88612b712ec5d41e4781d130bab):

Fatal error:

```
panic: priority range ID already set: old=2, new=64, first set at:

```
Stack: 

```
goroutine 100228 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc007590478, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc005802180, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x9b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc001fa8d80, {0xa722ca8, 0xc0008787e0}, 0xc00482ad90)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc001fa8d80, {0x1, 0xa, {0xa, 0x0, 0xc00482ad90, 0xc0075be9b0, 0xc00aa81b20, 0xc00aa817a0, 0xc002cd9040, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x2ae
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc00482ad90, 0xc0075be9b0, 0xc00aa81b20, 0xc00aa817a0, 0xc002cd9040, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1a5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc009cd6a80, {0xa722ca8, 0xc005f78ea0}, 0xc002f1b320)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x1f65
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc00359b500, {0xa722ca8, _}, {_, _}, {_, _}, {_, _}, {0xa6fc960, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0x167a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc006b98000, {0xa722c38, 0xc0001b4008})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x33db
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc001378000, {0xa722c38, 0xc0001b4008})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x90
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0xa6e65e0, 0xc008c06300}, {0xa6e5c20, 0xc0080a80a0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc001c70000)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x477
testing.tRunner(0xc001c70000, 0x7727ce8)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=64, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries2133129657
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109977 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1.10-rc release-blocker]
- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11629885?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11629885?buildTab=artifacts#/) on release-23.1 @ [6e34fffddbb0a88612b712ec5d41e4781d130bab](https://github.com/cockroachdb/cockroach/commits/6e34fffddbb0a88612b712ec5d41e4781d130bab):

Fatal error:

```
panic: priority range ID already set: old=2, new=64, first set at:

```
Stack: 

```
goroutine 99742 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x72
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc00474e518, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x9d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc000e5ee00, 0xb?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x9b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc00388b600, {0xa722ca8, 0xc003e00930}, 0xc0050320e0)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0xfc6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc00388b600, {0x1, 0xa, {0xa, 0x0, 0xc0050320e0, 0xc004968960, 0xc00a67d4f0, 0xc00a67d3e0, 0xc0083b3cc0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x2ae
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc0050320e0, 0xc004968960, 0xc00a67d4f0, 0xc00a67d3e0, 0xc0083b3cc0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1a5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc001f9b500, {0xa722ca8, 0xc00760a270}, 0xc004763e60)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x1f65
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc001dbaa80, {0xa722ca8, _}, {_, _}, {_, _}, {_, _}, {0xa6fc960, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0x167a
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc003c28000, {0xa722c38, 0xc0000820b8})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x33db
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc0070eec00, {0xa722c38, 0xc0000820b8})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x90
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0xa6e65e0, 0xc00427ec00}, {0xa6e5c20, 0xc0053990e0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x1b7
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc004255860)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x477
testing.tRunner(0xc004255860, 0x7727ce8)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
 [recovered]
	panic: priority range ID already set: old=2, new=64, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries414743930
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109977 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1.10-rc release-blocker]
- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestTxnCoordSenderRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11629883?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11629883?buildTab=artifacts#/) on release-23.1 @ [6e34fffddbb0a88612b712ec5d41e4781d130bab](https://github.com/cockroachdb/cockroach/commits/6e34fffddbb0a88612b712ec5d41e4781d130bab):

Fatal error:

```
panic: priority range ID already set: old=2, new=65, first set at:

```
Stack: 

```
goroutine 20422817 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc00176e010, 0x2)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:131 +0x35
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc002ba20a8?, 0xc01df55b30?)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/scheduler.go:321 +0x85
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc002ba2000, {0x5f3eae8, 0xc00ba2efc0}, 0xc0041c6770)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:403 +0x867
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).initRaftMuLockedReplicaMuLocked(0xc002ba2000, {0x1, 0xa, {0xa, 0x0, 0xc0041c6770, 0xc005d8f360, 0xc006119090, 0xc006119010, 0xc0170fe280, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:213 +0x1fd
github.com/cockroachdb/cockroach/pkg/kv/kvserver.newInitializedReplica(0x0?, {0x1, 0xa, {0xa, 0x0, 0xc0041c6770, 0xc005d8f360, 0xc006119090, 0xc006119010, 0xc0170fe280, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_init.go:67 +0x1c5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Start(0xc0157f3500, {0x5f3eae8, 0xc0093ad8f0}, 0xc024916580)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:1988 +0x114a
github.com/cockroachdb/cockroach/pkg/server.(*Node).start(0xc01bded500, {0x5f3eae8, _}, {_, _}, {_, _}, {_, _}, {0x5f18740, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/server/node.go:528 +0xfb9
github.com/cockroachdb/cockroach/pkg/server.(*Server).PreStart(0xc0150e0a80, {0x5f3ea78?, 0xc0001ac000?})
	github.com/cockroachdb/cockroach/pkg/server/server.go:1716 +0x1f76
github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start(0xc0069f7200, {0x5f3ea78, 0xc0001ac000})
	github.com/cockroachdb/cockroach/pkg/server/testserver.go:617 +0x36
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer({_, _}, {{{0x5f023e0, 0xc011a02c00}, {0x5f01a20, 0xc0068083c0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:323 +0x143
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestTxnCoordSenderRetries(0xc01bd81a00)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go:2016 +0x265
testing.tRunner(0xc01bd81a00, 0x4f34990)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
 [recovered]
	panic: priority range ID already set: old=2, new=65, first set at:
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnCoordSenderRetries
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestTxnCoordSenderRetries3431256497
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_cput_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_cput
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_initput
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_scan_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_put_and_get_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_put_and_scan_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_scan_and_cput
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_delete_range
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_cput_after_timestamp_leaked
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_and_put_after_timestamp_leaked
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_get_before_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_put_in_batch_commit
=== RUN   TestTxnCoordSenderRetries/forwarded_timestamp_with_too_many_refreshes,_read_only
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109977 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-23.1.10-rc release-blocker]
- #109873 kv/kvclient/kvcoord: TestTxnCoordSenderRetries failed [C-test-failure O-robot T-kv branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnCoordSenderRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
@@ -2011,6 +2011,8 @@ func TestTxnCoordSenderRetries(t *testing.T) {
 	// operates between keys "a" and "z" and expects a single split point at "b".
 	// See the call to setupMultipleRanges below.
 	storeKnobs.DisableLoadBasedSplitting = true
+	// Similarly, disable the merge queue to avoid unexpected merges.
+	storeKnobs.DisableMergeQueue = true
 
 	var refreshSpansCondenseFilter atomic.Value
 	s, _, _ := serverutils.StartServer(t,

